CalibTracker/SiPhase2TrackerESProducers/plugins/SiPhase2OuterTrackerFakeLorentzAngleESSource.cc

```diff
@@ -69,13 +69,15 @@ namespace fakeOTLA {
 
 std::unique_ptr<SiPhase2OuterTrackerLorentzAngle> SiPhase2OuterTrackerFakeLorentzAngleESSource::produceOTLA(
     const SiPhase2OuterTrackerLorentzAngleRcd& rcd) {
-  const auto& geomDet = rcd.getRecord<TrackerTopologyRcd>().get(m_geomDetToken);
+  const auto& geomDetRcd = rcd.getRecord<TrackerTopologyRcd>();
+  const auto& geomDet = geomDetRcd.get(m_geomDetToken);
```

Comment:
So the best course of action would be to
```suggestion
  const auto& geomDet = rcd.get(m_geomDetToken);
```
and to add `IdealGeometryRecord` is direct dependence of `SiPhase2OuterTrackerLorentzAngleRcd` and `SiPhase2OuterTrackerLorentzAngleSimRcd` in
https://github.com/cms-sw/cmssw/blob/3914bc83b3f9a0e3613f34c94bed4ec84f4b4cfe/CondFormats/DataRecord/interface/SiPhase2OuterTrackerCondDataRecords.h#L10-L15

Reply:
@iarspider I think this part of the suggestion was not applied:

```diff
-  const auto& geomDetRcd = rcd.getRecord<TrackerTopologyRcd>();
-  const auto& geomDet = geomDetRcd.get(m_geomDetToken);
+  const auto& geomDet = rcd.get(m_geomDetToken);
```